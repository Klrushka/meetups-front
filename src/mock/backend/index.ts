export * from './create.meetup.mock'
export * from './login.user.mock'
export * from './registrate.user.api'
export * from './delete.meetup.mock'
export * from './search.meetup.mock'
export * from './pdf.fetch.mock'
export * from './csv.fetch.mock'
