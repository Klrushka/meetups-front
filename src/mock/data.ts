export const data = [
  {
    _id: '633c2d65be13ecb347184b64',
    title: '1234567890-=',
    description:
      'jmgfdfgbcnfqwefqwaedaffafawfafafwawfawfcyrxtcyvubionuiyvktcjrxhercfygitfcrxdetrfyguiktfrdcfyguhktvcfrxdeszdxtcfgvykuivgtkcfdsgefqwefqwefqwfqwfqwfqwfwqfewqefwefwqefwfeqwfwqfwqfhjk.nlcridqwwdption',
    tags: ['T', 'A', 'g', 's'],
    userId: '633c29e825cb442183fae987',
    dueTime: '2022-06-10T09:34:41.552Z',
    location: {
      type: 'Point',
      coordinates: [27.563023567199707, 53.03491033436569],
      _id: '633c2d65be13ecb347184b65',
    },
    createdAt: '2022-10-04T12:56:05.618Z',
    updatedAt: '2022-10-04T12:56:05.618Z',
  },
  {
    _id: '633c34ea5900cea0c88570bc',
    title: '1234567890-=',
    description: 'descridqwwdption',
    tags: ['T', 'A', 'g', 's'],
    userId: '633c29e825cb442183fae987',
    dueTime: '2022-06-10T09:34:41.552Z',
    location: {
      type: 'Point',
      coordinates: [27.563023567199707, 53.03491033436569],
      _id: '633c34ea5900cea0c88570bd',
    },
    createdAt: '2022-10-04T13:28:10.820Z',
    updatedAt: '2022-10-04T13:28:10.820Z',
  },
  {
    _id: '633c35db5900cea0c88570bf',
    title: 'jhgvuy',
    description: 'dsf',
    tags: ['Tjioioiiom', 'A', 'g', 's', 'g', 'g', 'g', 'g', 'g', 'g', 'g'],
    userId: '633c29e825cb442183fae987',
    dueTime: '2022-06-10T09:34:41.552Z',
    location: {
      type: 'Point',
      coordinates: [27.563023567199707, 53.03491033436569],
      _id: '633c35db5900cea0c88570c0',
    },
    createdAt: '2022-10-04T13:32:11.875Z',
    updatedAt: '2022-10-04T13:32:11.875Z',
  },
  {
    _id: '633c35dd5900cea0c88570c2',
    title: 'jhgvuy',
    description: 'dsf',
    tags: ['Tjioioiiom', 'A', 'g', 's', 'g', 'g', 'g', 'g', 'g', 'g', 'g'],
    userId: '633c29e825cb442183fae987',
    dueTime: '2022-06-10T09:34:41.552Z',
    location: {
      type: 'Point',
      coordinates: [27.563023567199707, 53.03491033436569],
      _id: '633c35dd5900cea0c88570c3',
    },
    createdAt: '2022-10-04T13:32:13.905Z',
    updatedAt: '2022-10-04T13:32:13.905Z',
  },
  {
    _id: '633c35df5900cea0c88570c5',
    title: 'jhgvuy',
    description: 'dsf',
    tags: ['Tjioioiiom', 'A', 'g', 's', 'g', 'g', 'g', 'g', 'g', 'g', 'g'],
    userId: '633c29e825cb442183fae987',
    dueTime: '2022-06-10T09:34:41.552Z',
    location: {
      type: 'Point',
      coordinates: [27.563023567199707, 53.03491033436569],
      _id: '633c35df5900cea0c88570c6',
    },
    createdAt: '2022-10-04T13:32:15.773Z',
    updatedAt: '2022-10-04T13:32:15.773Z',
  },
  {
    _id: '633c35e15900cea0c88570c8',
    title: 'jhgvuy',
    description: 'dsf',
    tags: ['Tjioioiiom', 'A', 'g', 's', 'g', 'g', 'g', 'g', 'g', 'g', 'g'],
    userId: '633c29e825cb442183fae987',
    dueTime: '2022-06-10T09:34:41.552Z',
    location: {
      type: 'Point',
      coordinates: [27.563023567199707, 53.03491033436569],
      _id: '633c35e15900cea0c88570c9',
    },
    createdAt: '2022-10-04T13:32:17.407Z',
    updatedAt: '2022-10-04T13:32:17.407Z',
  },
  {
    _id: '633c38455900cea0c88570cb',
    title: 'qazxswecdcvr',
    description: 'dsf',
    tags: ['Tjioioiiom', 'A', 'g', 's', 'g', 'g', 'g', 'g', 'g', 'g', 'g'],
    userId: '633c29e825cb442183fae987',
    dueTime: '2022-06-10T09:34:41.552Z',
    location: {
      type: 'Point',
      coordinates: [27.563023567199707, 53.03491033436569],
      _id: '633c38455900cea0c88570cc',
    },
    createdAt: '2022-10-04T13:42:29.159Z',
    updatedAt: '2022-10-04T13:42:29.159Z',
  },
];
