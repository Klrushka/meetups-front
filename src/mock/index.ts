export * from './data';
export * from './navbar.routes';
export * from './routes';
