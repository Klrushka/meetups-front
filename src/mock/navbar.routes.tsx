export const navbarRoutes = [
  {value: 'PDF', to: '/pdf'},
  {value: 'Meetups', to: '/meetups'},
  {value: 'CSV', to: '/csv'},
]