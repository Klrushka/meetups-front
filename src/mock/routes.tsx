import React from 'react';
import { CsvPage } from '../pages/csv';
import { LoginPage } from '../pages/login';
import { MeetupPage } from '../pages/meetup';
import { PdfPage } from '../pages/pdf';
import { RegistratePage } from '../pages/registration';

export const routes = [
  { path: '/login', element: <LoginPage />},
  { path: '/meetups', element: <MeetupPage /> },
  { path: '/registration', element: <RegistratePage /> },
  { path: '/pdf', element: <PdfPage /> },
  { path: '/csv', element: <CsvPage /> },
];
