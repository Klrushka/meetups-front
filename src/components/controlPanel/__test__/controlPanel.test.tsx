import React from 'react';
import rendered, { act } from 'react-test-renderer';
import { ControlPanel } from '../index';
import { setIsShowMeetupRedactorWindow, setMeetupsMock, showModalMock } from './__mock__/mock.functions';

it('ControlPanel render', () => {
  let component: rendered.ReactTestRenderer;

  act(() => {
    component = rendered.create(
      <ControlPanel
        setShowModal={showModalMock}
        setMeetups={setMeetupsMock}
        meetups={[]}
        setIsShowMeetupRedactorWindow={setIsShowMeetupRedactorWindow}
      />
    );
  });

  expect(component.toJSON()).toMatchSnapshot();
});
