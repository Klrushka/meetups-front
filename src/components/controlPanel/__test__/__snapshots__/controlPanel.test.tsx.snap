// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlPanel render 1`] = `
<div
  className="sc-bcXHqe"
>
  <div>
    <div
      className="sc-gswNZR lpimJo"
    >
      <span
        className="sc-dkrFOg bPTfjF"
      >
         Your Meetups
      </span>
      <button
        className="sc-hLBbgP ikXIKy"
        onClick={[Function]}
      >
        New
      </button>
    </div>
    <div
      className="sc-eDvSVe kAekKi"
    >
      <input
        className="sc-jSUZER chnfnk"
        onChange={[Function]}
      />
    </div>
  </div>
</div>
`;
